text/entities/0256-entities-data-model.md

```diff
@@ -0,0 +1,790 @@
+
+<table>
+   <tr>
+    <td><strong>Field</strong>
+    </td>
+    <td><strong>Type</strong>
+    </td>
+    <td><strong>Description</strong>
+    </td>
+   </tr>
+   <tr>
+    <td>Type
+    </td>
+    <td>string
+    </td>
+    <td>Defines the type of the entity. MUST not change during the
+lifetime of the entity. For example: "service" or "host". This field is
+required and MUST not be empty for valid entities.
+    </td>
+   </tr>
+   <tr>
+    <td>Id
+    </td>
+    <td>map&lt;string, attribute&gt;
+    </td>
+    <td>Attributes that identify the entity.
+<p>
+MUST not change during the lifetime of the entity. The Id must contain
+at least one attribute.
+<p>
+Follows OpenTelemetry <a
+href="https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/common#attribute">common
+attribute definition</a>. SHOULD follow OpenTelemetry <a
+href="https://github.com/open-telemetry/semantic-conventions">semantic
+conventions</a> for attributes.
+    </td>
+   </tr>
+   <tr>
+    <td>Attributes
+    </td>
+    <td>map&lt;string, any&gt;
+    </td>
+    <td>Descriptive (non-identifying) attributes of the entity.
+<p>
+MAY change over the lifetime of the entity. MAY be empty. These
+attributes are not part of entity's identity.
+<p>
+Follows <a
+href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#type-any">any</a>
+value definition in the OpenTelemetry spec - it can be a scalar value,
+byte array, an array or map of values. Arbitrary deep nesting of values
+for arrays and maps is allowed.
+<p>
+SHOULD follow OpenTelemetry <a
+href="https://github.com/open-telemetry/semantic-conventions">semantic
+conventions</a> for attributes.
+    </td>
+   </tr>
+</table>
+
+### Minimally Sufficient Id
+
+Commonly, a number of attributes of an entity are readily available for the telemetry
+producer to compose an Id from. Of the available attributes the entity Id should
+include the minimal set of attributes that is sufficient for uniquely identifying
+that entity. No superfluous attributes should be included in the Id set. For example
+a Process on a host can be uniquely identified by (`process.pid`,`process.start_time`)
+attributes. Adding for example `process.executable.name` attribute to the Id is
+unnecessary and violates the Minimally Sufficient Id rule.
+
+### Examples of Entities
+
+_This section is non-normative and is present only for the purposes of demonstrating
+the data model._
+
+Here are examples of entities, the typical identifying attributes they
+have and some examples of non-identifying attributes that may be
+associated with the entity.
+
+<table>
+   <tr>
+    <td><strong>Entity</strong>
+    </td>
+    <td><strong>Entity Type</strong>
+    </td>
+    <td><strong>Identifying Attributes</strong>
+    </td>
+    <td><strong>Non-identifying Attributes</strong>
+    </td>
+   </tr>
+   <tr>
+    <td>Service
+    </td>
+    <td>"service"
+    </td>
```

Comment:
A service name is unique in a service namespace, k8S cluster, k8S namespace, deployment environment, etc.

Is one producer required to produce all those attributes - or can a producer say that the service name is unique in something else, e.g. k8S cluster?

Reply:
The [current convention](https://github.com/open-telemetry/semantic-conventions/tree/main/docs/resource#service) is that (service name, namespace, instance id) is unique. There is nothing defined about uniqueness of service names in k8s clusters, k8s namespaces, etc.

Otel SDK is required to produce service attributes that it knows about (normally the name and instance id). Collector can enrich the produced data by service namespace.

The exact details of what attributes are used for services (and other entity types) is out of this OTEP's scope and will be a future work.